receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go

```diff
@@ -27,20 +27,27 @@ const Type configmodels.Type = "hostmetricsreceiver"
 type metricIntf interface {
 	Name() string
 	New() pdata.Metric
+	Init(metric pdata.Metric)
 }
 
 // Intentionally not exposing this so that it is opaque and can change freely.
 type metricImpl struct {
-	name    string
-	newFunc func() pdata.Metric
+	name     string
+	initFunc func(pdata.Metric)
 }
 
 func (m *metricImpl) Name() string {
 	return m.name
 }
 
 func (m *metricImpl) New() pdata.Metric {
-	return m.newFunc()
```

Comment:
Maybe we should consider to remove the New (or deprecate it)

Reply:
Hmm, well the New is useful if you're not trying to optimize allocations and just do New/Appends.